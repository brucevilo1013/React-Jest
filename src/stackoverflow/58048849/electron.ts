export const ipcRenderer = {
  events: {},
  on(event, handler) {
    this.events[event] = handler;
  },
  send(event, data) {
    this.events[event](event, data);
  },
  removeAllListeners(event) {
    this.events[event] = undefined;
  }
};
