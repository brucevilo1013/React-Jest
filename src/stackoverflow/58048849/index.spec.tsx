import React from 'react';
import { render, act } from '@testing-library/react';
import { ipcRenderer } from './electron';
import AccountCheckModule from './';

describe('AccountCheckModule', () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });
  it('should render correct', async () => {
    const events = {};
    const onSpy = jest.spyOn(ipcRenderer, 'on').mockImplementation((event, handler) => {
      events[event] = handler;
    });
    const sendSpy = jest.spyOn(ipcRenderer, 'send').mockImplementation((event, data) => {
      events[event](event, data);
    });
    const { getByText, container } = render(<AccountCheckModule></AccountCheckModule>);
    const mCount = 666;
    act(() => {
      ipcRenderer.send('count-listings', mCount);
    });
    const element = getByText(mCount.toString());
    expect(element).toBeDefined();
    expect(onSpy).toBeCalledWith('count-listings', expect.any(Function));
    expect(sendSpy).toBeCalledWith('count-listings', mCount);
    expect(container).toMatchSnapshot();
  });
});
