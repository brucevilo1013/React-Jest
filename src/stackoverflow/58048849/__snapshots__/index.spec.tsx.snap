// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountCheckModule should render correct 1`] = `
<div>
  <div>
    666
  </div>
</div>
`;
